File: src/main/java/se/clouds/app/javanet/app/domain/feature/Feature.java
package se.clouds.app.javanet.app.domain.feature;

import java.util.HashMap;
import java.util.Map;

public class Feature
{
    public String Name;
    public boolean Activated;
    public String Description;

    public Feature(String name, boolean activated, String description) {
        Name = name;
        Activated = activated;
        Description = description;
    }

    public String getName() {
        return Name;
    }

    public void setName(String name) {
        Name = name;
    }

    public boolean isActivated() {
        return Activated;
    }

    public void setActivated(boolean activated) {
        Activated = activated;
    }

    public String getDescription() {
        return Description;
    }

    public void setDescription(String description) {
        Description = description;
    }

    public Map<String, Object> AsMap(){
        Map<String, Object> map = new HashMap<String, Object>();
        map.put("name", Name);
        map.put("activated", Activated);
        map.put("description", Description);
        return map;
    }
}

File: src/main/dotnet/app/domain/feature/Feature.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using se.clouds.javanet.core.domain;

namespace se.clouds.javanet.app.domain.feature
{
    public class Feature : AggregateRoot<Feature>
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        [JsonIgnore]
        public Guid Id { get; set; }

        public bool Activated { get; set; }

        public string Version { get; set; }

        public string Description { get; set; }

        public override string ToString()
        {
            return string.Format($"Feature (Name = {Name}, Activated = {Activated})");
        }

        protected override bool EqualsCore(Feature other)
        {
            throw new NotImplementedException();
        }

        protected override int GetHashCodeCore()
        {
            throw new NotImplementedException();
        }
    }
}
